bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/build/PayloadCodingRegistry.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat Inc and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.packagedrone.utils.rpm.build;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class PayloadCodingRegistry
```

Comment:
Since the class instance needs to be looked up based on the string value in the RPM header, some way is needed to look up the class instance. I couldn't prevent a new class instance from being created manually, although default instances that need gzip (such as in `PayloadRecorder`) are looking it up in the registry.